src/test/java/com/datadog/api/v2/client/api/SecurityMonitoringApiTest.java

```diff
@@ -0,0 +1,171 @@
+
+        Set<String> idsIntersection = new HashSet<>(firstPageIds);
+        idsIntersection.retainAll(secondPageIds);
+        assertEquals(0, idsIntersection.size());
+
+        //// update rule
+        SecurityMonitoringRuleResponse updatedRule = api.updateSecurityMonitoringRule(createdRule.getId())
+                .body(new SecurityMonitoringRuleUpdatePayload()
+                        .name(createdRule.getName())
+                        .enabled(false)
+                        .queries(createdRule.getQueries())
+                        .options(createdRule.getOptions())
+                        .cases(createdRule.getCases())
+                        .message(createdRule.getMessage())
+                        .tags(createdRule.getTags()))
+                .execute();
+        assertEquals(createdRule.getName(), updatedRule.getName());
+        assertEquals(false, updatedRule.getIsEnabled());
+
+        SecurityMonitoringRuleResponse getUpdatedRule = api.getSecurityMonitoringRule(createdRule.getId()).execute();
+        assertEquals(false, getUpdatedRule.getIsEnabled());
+
+        //// delete rule
+        api.deleteSecurityMonitoringRule(createdRule.getId()).execute();
+        try {
+            api.getSecurityMonitoringRule(createdRule.getId()).execute();
+        } catch (ApiException e) {
+            assertEquals(404, e.getCode());
+        }
+    }
+
+    private SecurityMonitoringRuleResponse createRule(String ruleName) throws ApiException {
+        SecurityMonitoringRuleCreatePayload createRulePayload = new SecurityMonitoringRuleCreatePayload();
+        createRulePayload
+                .name(ruleName)
+                .enabled(true)
+                .addQueriesItem(new SecurityMonitoringRuleQuery()
+                        .name("nevermatch")
+                        .query("thiswillnevernevermatch")
+                        .groupByFields(Collections.emptyList()))
+                .options(new SecurityMonitoringRuleOptions()
+                        .evaluationWindow(SecurityMonitoringRuleEvaluationWindow.FIVE_MINUTES)
+                        .keepAlive(SecurityMonitoringRuleKeepAlive.FIVE_MINUTES)
+                        .maxSignalDuration(SecurityMonitoringRuleMaxSignalDuration.FIVE_MINUTES))
+                .addCasesItem(new SecurityMonitoringRuleCase()
+                        .name("rule-case")
+                        .condition("nevermatch > 1000")
+                        .status(SecurityMonitoringRuleSeverity.INFO))
+                .message("Rule message")
+                .tags(Collections.singletonList("datadog-api-client-test-java"));
+
+        return api.createSecurityMonitoringRule()
+                .body(createRulePayload)
+                .execute();
+    }
+
+    private String getUniqueSuffix() {
+        String now = DateTimeFormatter.ISO_LOCAL_DATE_TIME.format(LocalDateTime.now());
+        return "-" + now + "-" + UUID.randomUUID();
```

Comment:
The random UUID will not work with recorded cassettes unless we store a seed during recording (example for datadogpy https://github.com/DataDog/datadogpy/blob/master/tests/integration/dogshell/test_dogshell.py#L118-L144).